pcdsdevices/state.py

```diff
@@ -72,16 +72,15 @@ def __init__(self, prefix, *, name, **kwargs):
         if self.__class__ is StatePositioner:
             raise TypeError(('StatePositioner must be subclassed with at '
                              'least a state signal'))
-        super().__init__(prefix, name=name, **kwargs)
         self._state_initialized = False
-        self._state_init_cbid = False
+        super().__init__(prefix, name=name, **kwargs)
         if self.states_list:
             self._state_init()
-        else:
-            cbid = self.state.subscribe(self._late_state_init,
-                                        event_type=self._state_meta_sub,
-                                        run=False)
-            self._state_init_cbid = cbid
+
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
+        if cls.state is not None and not cls.states_list:
+            cls.state.sub_meta(cls._late_state_init)
```

Comment:
To expand on what's going on here (for lurkers):
Ken added these useful subscription decorators a while back, with intended usage like:
```
class MyClass(Parent):
    state = Cpt(EpicsSignal, 'PVNAME')

    @state.sub_meta
    def func_to_subscribe_at_init(self):
        ...
```

but I can't use them the normal way because the `state` signal is not defined yet in the parent states class. Instead, I apply them in `__init_subclass__`, which is run when the subclass is created, to schedule the subscription callback onto whatever `state` signal the user has configured for the device.